src/modal-checkout/modal.js

```diff
@@ -49,40 +49,53 @@ domReady( () => {
 
 	const initialHeight = modalContent.clientHeight + spinner.clientHeight + 'px';
 	const closeCheckout = () => {
-		spinner.style.display = 'flex';
-
 		const container = iframe?.contentDocument?.querySelector( `#${ IFRAME_CONTAINER_ID }` );
-
-		if ( iframe && modalContent.contains( iframe ) ) {
-			// Reset iframe and modal content heights.
-			iframe.src = 'about:blank';
-			iframe.style.height = '0';
-			modalContent.removeChild( iframe );
-			modalContent.style.height = initialHeight;
+		const afterSuccessUrlInput = container.querySelector( 'input[name="after_success_url"]' );
+		const afterSuccessBehaviorInput = container.querySelector(
+			'input[name="after_success_behavior"]'
+		);
+		const hasNewsletterPopup = document?.querySelector( '.newspack-newsletters-signup-modal' );
+
+		// We want to block closing the modal if redirecting elsewhere:
+		let shouldCloseModal = true;
+
+		if (
+			afterSuccessUrlInput &&
+			afterSuccessBehaviorInput &&
+			container?.checkoutComplete &&
+			! hasNewsletterPopup
+		) {
+			shouldCloseModal = false;
 		}
 
-		if ( iframeResizeObserver ) {
-			iframeResizeObserver.disconnect();
-		}
+		if ( shouldCloseModal ) {
```

Comment:
If you decide to go with the strategy above for `shouldCloseModal`:

```suggestion
		if ( shouldCloseModal || hasNewslettersPopup ) {
```